src/main/java/org/opentripplanner/routing/algorithm/transferoptimization/services/TransitPathLegSelector.java

```diff
@@ -28,38 +28,35 @@
  */
 class TransitPathLegSelector<T extends RaptorTripSchedule> {
 
-  private final MinCostFilterChain<OptimizedPathTail<T>> filter;
+  private final PathTailFilter<T> filter;
   private Set<OptimizedPathTail<T>> remindingLegs;
   private Set<OptimizedPathTail<T>> selectedLegs;
 
-  private int lastLimit = Integer.MAX_VALUE;
+  private int prevStopPosition = Integer.MAX_VALUE;
 
-  TransitPathLegSelector(
-    final MinCostFilterChain<OptimizedPathTail<T>> filter,
-    final Set<OptimizedPathTail<T>> legs
-  ) {
+  TransitPathLegSelector(final PathTailFilter<T> filter, final Set<OptimizedPathTail<T>> legs) {
     this.filter = filter;
     this.remindingLegs = Set.copyOf(legs);
     this.selectedLegs = new HashSet<>();
   }
 
-  Set<OptimizedPathTail<T>> next(final int earliestBoardingTime) {
-    if (earliestBoardingTime > lastLimit) {
+  Set<OptimizedPathTail<T>> next(final int fromStopPosition) {
+    if (fromStopPosition > prevStopPosition) {
       throw new IllegalStateException(
         "The next method must be called with decreasing time limits. " +
-        "minTimeLimit=" +
-        TimeUtils.timeToStrLong(earliestBoardingTime) +
-        ", lastLimit=" +
-        TimeUtils.timeToStrLong(lastLimit)
+        "fromStopPosition=" +
+        fromStopPosition +
+        ", previousStopPosition=" +
+        prevStopPosition
```

Comment:
I think we should have a quick chat in one of the dev meetings how we prefer to build template strings.

I personally prefer the `"Hello %s".formatted("World")` style but I'm not sure what others think.

Reply:
I think it depends. When handling exceptions we want to be extra careful not to do mistakes. The String concatenation(`+`) is regarded as the safest way to create Strings - and in many cases it is recommended to use it and not e.g. string builder. Using String.format() is nice if we intend to format the information, but in exceptions we do not want ANY transformations. Also it is easy do do mistakes with the `String.format()`. ToStrings are easy to test with a unit test, while we error branches are frequently not tested.